enhancements/manifestlist/manifestlist-support.md

```diff
@@ -0,0 +1,466 @@
+
+```go
+type ImageImportStatus struct {
+	Status metav1.Status
+	Image *Image
+	Tag string
+}
+```
+
+`ImageImportStatus` struct has a reference for a single image on its `Image` field. This
+enhancement proposes the addition of another field where child images (manifests within a
+manifes list) would be listed, leaving `ImageImportStatus` struct looking like:
+
+
+```go
+type ImageImportStatus struct {
+	Status metav1.Status
+	Image *Image
+	Tag string
+	Manifests []Image
+}
+```
+
+Here the image that refers to the manifest list would be presented on `Image` property (as it is
+today for regular manifests) while all inner manifests within the manifest list would be listed
+in the `Manifests` slice.
+
+### Proposed ImageStreamImage changes
+
+ImageStreamImages are a way of referring to a specific Image within an ImageStream by its hash.
+It is useful when tagging an ImageStreamTag into another ImageStreamTag. With ImageStreamImage
+one can issue a command such as:
+
+```console
+$ oc tag imagestream0:tag0 imagestream1:tag1
+```
+
+This command would create a tag similar to this one inside "imagestream1":
+
+```yaml
+spec:
+  tags:
+  - name: tag1
+    from:
+      kind: ImageStreamImage
+      name: imagestream0@sha256:<sha256>
+```
+
+ImageStreamImage can also be used to "query details" about a specific Image, by using the
+following command a user can obtain information about a specific Image inside an ImageStream:
+
+```console
+$ oc get isimage <imagestream>@<sha of one of the tags>
+```
+
+With the introduction of the manifest list support, there won't be many details about an Image
+on its "parent" manifest (extra information belongs to child images). This enhancement proposes
+an extension to allow users to also view details for any of the child images.
+
+Both of the commands below would be valid, the first command would not return that much
+information but it works as a means of obtaining a list of all child images:
+
+```console
+$ oc get isimage <imagestream>@<sha of the parent image>
+$ oc get isimage <imagestream>@<sha of any of the child images>
+```
+
+The output of the `oc describe isimage` command line will need to be changed to display
+information present in the DockerImageManifests property of the parent Image.
+
+### Proposed pruner changes
+
+Pruner gathers information about images in use within a cluster by inspecting the following
+objects:
+
+- Pods
+- Replication controllers
+- Deployment configs
+- Replica sets
+- Deployments
+- Daemon sets
+- Builds
+- Build configs
+- Stateful sets
+- Jobs
+- Cronjobs
+
+If a given image is in use by any of these entities it is not pruned. This operation will need
+to be refactored to take into account the extra level of indirection that the "child images"
+layer brings. For example, in a scenario where we have the following:
+
+```text
+ImageStream X -> Image A -> Image B
+                            Image C
+                            Image D
+
```

Comment:
it's not clear to me what the relationship is between A,B,C,and D here.

are they all part of the same manifestlist?  is B the base image for A?  Is A a manifestlist that contains B,C, and D?

I think it's that last option, but it'd help to clarify the scenario.

Reply:
Yes, you are right. I will clarify it. The last option is the correct one.